cypress/integration/donation.spec.js

```diff
@@ -0,0 +1,53 @@
+import faker from 'faker'
+
+describe('Donation Spec', () => {
+  beforeEach(() => {
+    cy.request('/cypress_rails_reset_state')
+  })
+
+  it('renders donation widget', () => {
+    cy.visit('/test/widget/donation')
+    cy.contains('DONATE', { matchCase: false })
+  })
+
+  it('makes a donation', () => {
```

Comment:
How this test is different from the ones we already have?
It seems you are using a mock backend endpoint too, or I am wrong?

Reply:
Yeah, we are using the actual endpoint here, this is configured here. 

This is an integration test, so if you are mocking the endpoint probably the tests will look the same. There's no easy way to setup test cases without having access to the backend.

https://github.com/debtcollective/membership/pull/373/files#diff-3e7704557ff348aeb70b9950524b0d1924b85cbc6b567d2fe209596d22de1b4dR18-R22

Reply:
Then we are doing a proper e2e not an integration test.
- **Integration:** Verify that several units work together in harmony. Mocking a little as possible often times requests
- **E2E:** Typically these will run the entire application (both frontend and backend) and your test will interact with the app just like a typical user would. Sometimes called "functional testing" or e2e.